mountpoint-s3/src/cli.rs

```diff
@@ -0,0 +1,32 @@
+use crate::build_info;
+
+use clap::Parser;
+use mountpoint_s3_fs::cli::CliArgs;
+use std::env;
+use std::ffi::OsString;
+
+use crate::fstab::FsTabCliArgs;
+
+// TODO: Extract `mountpoint_s3_fs::cli::CliArgs` to this crate.
+#[derive(Parser, Debug)]
+#[clap(
+    name = "mount-s3",
+    about = "Mountpoint for Amazon S3",
+    version = build_info::FULL_VERSION,
+)]
+struct AppCliArgs {
+    #[clap(flatten)]
+    pub cli_args: CliArgs,
+}
+
+pub fn get_cli_args() -> anyhow::Result<CliArgs> {
+    let is_fstab = env::args_os().len() == 5 && env::args_os().nth(3) == Some(OsString::from("-o"));
+
+    if is_fstab {
+        let args = FsTabCliArgs::parse();
```

Comment:
It's probably very unlikely but – should we print something like "fstab mode detected" in this block to ensure people running into this block accidentally can quickly understand what's going on?